### PR TITLE
CSHARP-4728: Add implementation for Atlas Search Sort

### PR DESCRIPTION
**Pull request by sboulema**
The 10 July 2023 release of MongoDB Atlas Search introduced the sort option.

[Atlas Search Changelog - 10 July 2023 Release](https://www.mongodb.com/docs/atlas/atlas-search/changelog/#10-july-2023-release)

[Sort Atlas Search Results](https://www.mongodb.com/docs/atlas/atlas-search/sort/#std-label-sort-ref)

It would be very nice to have access to the sort field from the C# driver

### CODE REVIEW

File: tests/MongoDB.Driver.Tests/Search/AtlasSearchTests.cs [hunk 1]
```diff
@@ -112,6 +112,19 @@ public void Count_total()
...
+        public void Sort()
+        {
+            var results = GetTestCollection().Aggregate()
+                .Search(
+                    Builders.Search.Text(x => x.Body, "liberty"),
+                    sort: Builders.Sort.Descending(x => x.Title))
+                .Project<HistoricalDocument>(Builders.Projection.Include(x => x.Title))
+                .Limit(1)
+                .ToList();
+            results.Should().ContainSingle().Which.Title.Should().Be("Gettysburg Address");
```
Comment (BorisDog) [inline tests/MongoDB.Driver.Tests/Search/AtlasSearchTests.cs:125]: Small modification:

```
  var results = GetTestCollection().Aggregate()
                .Search(
                    Builders.Search.Text(x => x.Body, "liberty"),
                    sort: Builders.Sort.Descending(x => x.Title))
                .Project<HistoricalDocument>(Builders.Projection.Include(x => x.Title))
                .Limit(1)
                .ToList();
            results.Should().ContainSingle().Which.Title.Should().Be("US Constitution");
```

**Review by BorisDog — CHANGES_REQUESTED** [review]
Thanks, looks great overall! Few small comments.

File: tests/MongoDB.Driver.Tests/Search/AtlasSearchTests.cs [hunk 2]
```diff
@@ -402,6 +402,19 @@ public void SearchMeta_facet()
             bucket.Count.Should().Be(108);
         }
 
+        [Fact]
+        public void Sort()
```
Comment (BorisDog) [inline tests/MongoDB.Driver.Tests/Search/AtlasSearchTests.cs:406]: Thanks, please place `Sort` after `Should`.

**Review by BorisDog — CHANGES_REQUESTED** [review]

File: tests/MongoDB.Driver.Tests/Search/AtlasSearchTests.cs (hunk 2 above)
Comment (sboulema) [reply tests/MongoDB.Driver.Tests/Search/AtlasSearchTests.cs:406]: Wow I feel stupid :|

File: tests/MongoDB.Driver.Tests/Search/AtlasSearchTests.cs [hunk 3]
```diff
@@ -413,6 +413,19 @@ public void Should()
...
+        {
+            var results = GetTestCollection().Aggregate()
+                .Search(
+                    Builders.Search.Text(x => x.Body, "liberty"),
+                    sort: Builders.Sort.Descending(x => x.Title))
+                .Project<HistoricalDocument>(Builders.Projection.Include(x => x.Title))
+                .Limit(1)
+                .ToList();
+            results.Should().ContainSingle().Which.Title.Should().Be("US Constitution");
```
Comment (BorisDog) [inline tests/MongoDB.Driver.Tests/Search/AtlasSearchTests.cs:426]: Thanks, looks good now.
One last thing, due to sort indexes changes on our side,  `MustNot` test fails.
Can you please change 
`result.Title.Should().Be("US Constitution")`
to 
`result.Title.Should().Be("Gettysburg Address")`

in `MustNot` test?

**Review by BorisDog — APPROVED** [review]
LGTM

**Review by sanych-sun — APPROVED** [review]
LGTM

**Pull request merged**